cpp/common/env.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) ZeroC, Inc.
+
+#include <optional>
+#include <string>
+
+namespace common
+{
+    /// Get the value of an environment variable.
+    /// @param variableName The name of the environment variable.
+    /// @return The value of the environment variable, or nullopt if the variable is not set.
+    std::optional<std::string> getEnvironmentVariable(const char* variableName)
+    {
+#ifdef _MSC_VER
+        size_t requestedSize;
+        char buffer[256];
+        if (getenv_s(&requestedSize, buffer, sizeof(buffer), variableName))
+        {
+            // getenv_s failed, set requestedSize to 0 to indicate that the buffer is not valid.
```

Comment:
I am not fond of this code, looks hacky.

Reply:
Fixed.